`npm run typecheck` fails in npx create-remix@2.5.1 with Type 'FSWatcher' is missing the following properties from type 'FSWatcher': ref, unref

### Reproduction

* `npx create-remix@2.5.1`
* `npm run typecheck`

```
node_modules/vite/dist/node/index.d.ts:68:15 - error TS2420: Class 'import("/home/justintoman/code/personal/remix-test/node_modules/vite/dist/node/index").FSWatcher' incorrectly implements interface 'import("fs").FSWatcher'.
  Type 'FSWatcher' is missing the following properties from type 'FSWatcher': ref, unref

68 declare class FSWatcher extends EventEmitter implements fs.FSWatcher {
                 ~~~~~~~~~


Found 1 error in node_modules/vite/dist/node/index.d.ts:68
```

I don't know if this is Remix's fault... but it does seem like Remix's problem. I also don't understand why the non-Vite template depends on Vite.

### System Info

```shell
I tried this once on WSL and once on MacOS

WSL:
  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (32) x64 13th Gen Intel(R) Core(TM) i9-13950HX
    Memory: 10.28 GB / 15.62 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
    bun: 1.0.22 - ~/.bun/bin/bun
  npmPackages:
    @remix-run/css-bundle: ^2.5.1 => 2.5.1
    @remix-run/dev: ^2.5.1 => 2.5.1
    @remix-run/node: ^2.5.1 => 2.5.1
    @remix-run/react: ^2.5.1 => 2.5.1
    @remix-run/serve: ^2.5.1 => 2.5.1 


MacOS:

System:
    OS: macOS 14.2.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 17.56 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.21 - ~/.nvm/versions/node/v20.10.0/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.10.0/bin/npm
    pnpm: 8.11.0 - ~/Library/pnpm/pnpm
    bun: 1.0.18 - ~/.bun/bin/bun
  Browsers:
    Chrome: 120.0.6099.234
    Safari: 17.2.1
  npmPackages:
    @remix-run/css-bundle: ^2.5.1 => 2.5.1
    @remix-run/dev: ^2.5.1 => 2.5.1
    @remix-run/node: ^2.5.1 => 2.5.1
    @remix-run/react: ^2.5.1 => 2.5.1
    @remix-run/serve: ^2.5.1 => 2.5.1
```


### Used Package Manager

npm

### Expected Behavior

tsc should work when creating a new app from scratch

### Actual Behavior

Type error

--- a/templates/arc/tsconfig.json
+++ b/templates/arc/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/cloudflare-pages/tsconfig.json
+++ b/templates/cloudflare-pages/tsconfig.json
@@ -10,12 +10,12 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {
       "~/*": ["./app/*"]
     },
-    "skipLibCheck": true,
 
     // Remix takes care of building everything in `remix build`.
     "noEmit": true

--- a/templates/cloudflare-workers/tsconfig.json
+++ b/templates/cloudflare-workers/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/express/tsconfig.json
+++ b/templates/express/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/fly/tsconfig.json
+++ b/templates/fly/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/remix-tutorial/tsconfig.json
+++ b/templates/remix-tutorial/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/remix/tsconfig.json
+++ b/templates/remix/tsconfig.json
@@ -10,6 +10,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

--- a/templates/spa/tsconfig.json
+++ b/templates/spa/tsconfig.json
@@ -11,6 +11,7 @@
     "target": "ES2022",
     "strict": true,
     "allowJs": true,
+    "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
     "baseUrl": ".",
     "paths": {

